src/calendar.cpp

```diff
@@ -257,11 +257,33 @@ bool calendar::is_night() const
     return (seconds > sunset_seconds + TWILIGHT_SECONDS || seconds < sunrise_seconds);
 }
 
+double calendar::current_daylight_level() const
+{
+    double current_daylight_modifier = 1.;
+    
+    switch (season) {
+    case SPRING:
+        current_daylight_modifier = 1.;
+        break;
```

Comment:
@Coolthulhu thinks the performance impact is small, so I implemented the smooth changes you suggested.